File: web/src/types/delivery.ts
// Delivery configuration types for workflow notifications

export type DeliveryChannelType = 'slack' | 'line';

export type DeliveryChannelStatus = 'pending' | 'connected' | 'error';

export interface SlackChannel {
  id: string;
  name: string;
}

export interface SlackDeliveryChannel {
  type: 'slack';
  channelId: string;
  channelName: string;
  status: DeliveryChannelStatus;
  workspaceId?: string;
  workspaceName?: string;
}

export interface LineDeliveryChannel {
  type: 'line';
  status: DeliveryChannelStatus;
  notifyToken?: string;
}

export type DeliveryChannel = SlackDeliveryChannel | LineDeliveryChannel;

export interface DeliveryConfig {
  channels: DeliveryChannel[];
  includeAiSummary: boolean;
}

export const DEFAULT_DELIVERY_CONFIG: DeliveryConfig = {
  channels: [],
  includeAiSummary: false,
};

// Execution delivery result — stored per execution run
export type ChannelDeliveryStatus = 'delivered' | 'failed' | 'not_configured';

export interface ExecutionDeliveryResult {
  channelType: DeliveryChannelType;
  status: ChannelDeliveryStatus;
  channelLabel?: string; // e.g. "#marketing-alerts"
  error?: string;
}

File: common/common/model/delivery.py
from enum import StrEnum

from pydantic import BaseModel


class DeliveryChannel(StrEnum):
    SLACK = "slack"
    LINE = "line"


class SlackChannelConfig(BaseModel):
    channel: DeliveryChannel = DeliveryChannel.SLACK
    channel_id: str
    channel_name: str
    bot_token: str


class LineChannelConfig(BaseModel):
    channel: DeliveryChannel = DeliveryChannel.LINE
    access_token: str
    to: str


class DeliveryConfig(BaseModel):
    channels: list[SlackChannelConfig | LineChannelConfig]
    include_ai_summary: bool = False
